opensync-gateway/src/main/java/com/telecominfraproject/wlan/opensync/mqtt/OpensyncMqttClient.java

```diff
@@ -188,27 +190,14 @@ public void run() {
 
                         // main loop - receive messages
                         while (true) {
-                            Message mqttMsg = futureConnection.receive().await();
-
-                            if (mqttMsg == null) {
-                                continue;
-                            }
+                            
+                            Future<Message> receive = futureConnection.receive();
+                            Message mqttMsg = receive.await();
+                            mqttMsg.ack();
```

Comment:
Why remove the null check before acknowledging the message here?

Reply:
Wrong file, thanks for noticing, need to name them better in the future. See the 2nd commit please